src/main/kotlin/com/terraformation/backend/accelerator/EventNotifier.kt

```diff
@@ -0,0 +1,52 @@
+package com.terraformation.backend.accelerator
+
+import com.terraformation.backend.accelerator.db.ModuleStore
+import com.terraformation.backend.accelerator.event.ModuleEventScheduledEvent
+import com.terraformation.backend.accelerator.event.ModuleEventStartingEvent
+import com.terraformation.backend.customer.model.SystemUser
+import com.terraformation.backend.log.perClassLogger
+import java.time.Duration
+import java.time.InstantSource
+import javax.inject.Named
+import org.jobrunr.scheduling.JobScheduler
+import org.springframework.context.ApplicationEventPublisher
+import org.springframework.context.annotation.Lazy
+import org.springframework.context.event.EventListener
+
+@Named
+class EventNotifier(
+    private val clock: InstantSource,
+    private val eventPublisher: ApplicationEventPublisher,
+    @Lazy private val scheduler: JobScheduler,
+    private val moduleStore: ModuleStore,
+    private val systemUser: SystemUser,
+) {
+  companion object {
+    /** Notify upcoming events with the given lead time. */
+    private val notificationLeadTime = Duration.ofMinutes(15)
+    private val log = perClassLogger()
+  }
+
+  /** Schedules an "event is starting" notification according to preset time. */
+  @EventListener
+  fun on(event: ModuleEventScheduledEvent) {
+    val moduleEvent = moduleStore.fetchEventById(event.eventId)
+    if (moduleEvent?.startTime != null) {
+      val notifyTime = moduleEvent.startTime.minus(notificationLeadTime)
+      if (notifyTime.isAfter(clock.instant())) {
```

Comment:
Per requirement in design, we are choosing to not show notifications if event changes is happening within the next 15 minutes. 

Likely accelerator admins will be warned before making changes to events too soon or in the past (in the admin UI), and will be responsible for contacting attendants outside of Terraware should that happen.